pkg/reconciler/gc/controller.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gc
+
+import (
+	"context"
+	"time"
+
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	configurationinformer "knative.dev/serving/pkg/client/injection/informers/serving/v1alpha1/configuration"
+	revisioninformer "knative.dev/serving/pkg/client/injection/informers/serving/v1alpha1/revision"
+	pkgreconciler "knative.dev/serving/pkg/reconciler"
+	configns "knative.dev/serving/pkg/reconciler/configuration/config"
+)
+
+const (
+	controllerAgentName = "gc-controller"
+	resyncInterval      = 4 * time.Hour
```

Comment:
Nit: I think its fine for us to decrease this value but with a change like this I prefer to keep our existing behavior first and then update configuration like this in a later change - makes it easier to notice if we broke anything with this change. Currently I believe this value is 10 * time.Hour.

Reply:
Actually, rather than statically set this to 10, can we keep using controller.GetResyncPeriod(ctx)?

Reply:
@greghaynes I was talking to @vagababov, he mentioned he had a discussion with @dgerd and decided to resync every hour here. @vagababov what's the reasoning behind it?

Reply:
@greghaynes I was talking to @vagababov, he mentioned he had a discussion with @dgerd and decided to resync every hour here. @vagababov what's the reasoning behind it?

Reply:
Dan's point was that if users set their GC collection time to a short window, e.g. 1 hour, keeping them around for 10 hours is not very nice.

Reply:
Taking a look at `config-gc.yaml` it looks like we prevent (or maybe just highly encourage) stale-revision-timeout from going lower than the resync period. 

```
    # Duration since a route has been pointed at a revision before it should be GC'd
    # This minus lastpinned-debounce be longer than the controller resync period (10 hours)
    stale-revision-timeout: "15h"
```

We likely will want to make this at least twice as fast as the stale-revision-timeout to avoid having it just miss the resync, but I am going to agree with greg on this to keep it the same for this change and follow up with lowering it.